lib/main.dart

```diff
@@ -1,6 +1,38 @@
 import 'package:flutter/material.dart';
+import 'dart:convert';
+import 'dart:async';
+import 'package:http/http.dart' as http;
+import 'package:flutter/foundation.dart';
+import 'catData.dart';
 
 void main() {
-  runApp(MyApp());
+  runApp(MainPage());
 }
 
+Future<List<CatData>> fetchCats(http.Client client) async {
+  final response = await client.get('https://api.thecatapi.com/v1/breeds/');
+  return compute(parseCat, response.body);
+}
+
+List<CatData> parseCat(responseBody) {
```

Comment:
Try to have some structure in the project and keep  main.dart as simple and empty as possible.

Reply:
fixed